pulsar-client-api/src/main/java/org/apache/pulsar/client/api/schema/Field.java

```diff
@@ -40,4 +33,42 @@
      */
     private final int index;
 
+    public Field(String name, int index) {
```

Comment:
Why not use Lombok? It will not be required at runtime anyway

Reply:
Then I will have to pull in the lombok dependency.  Didn't feel appropriate.  Trying to keep pulsar-client-api as minimal as possible